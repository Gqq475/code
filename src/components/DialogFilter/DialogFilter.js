import React from 'react'
import classes from './DialogFilter.scss'
import { Button, Dialog } from '@blueprintjs/core'

type Props = {
  data: Obejct,
  isOpen: boolean,
  DialogTitle: String,
  DialogID: Number,
  DialogName:String,
  modelId: Number,
  DialogBtnContent: String,
  toggleDialog: Function,
  handleSubmit: Function,
  that:Object,
  kernelId: Number,
  kernelIds: Object,
  loadingKernelIds: Boolean
};
export class DialogFilter extends React.Component {
  props: Props;

  constructor (props) {
    super(props)
    this.handleChange = this.handleChange.bind(this)
    this.isChanged = this.isChanged.bind(this)
  }

  // handleChange (type) {
  //   return (e) => {
  //     this.props.handleChange(e.target.value, type)
  //   }
  // }

  handleChange (type) {
    return (e) => {
      this.props.that.setState({
        [type]: this.trim(e.target.value)
      })
    }
  }

  trim (str) {
    return str.replace(/(^\s*)|(\s*$)/g, '')
  }

  isChanged () {
    return (((this.refs.serviceName && this.trim(this.refs.serviceName.value)) || this.props.DialogName) &&
      (+this.props.kernelId || +this.props.kernelId === 0) &&
      +this.props.modelId)
  }

  render () {
    return (
      <div className={classes['DialogFilter-container']}>
        <Dialog
          isOpen={this.props.isOpen}
          onClose={this.props.toggleDialog}
          className={classes.dialog}
        >
          <div className={classes.header}>
            <h6 className={this.props.DialogID === '' ? classes.onlyHeaderTitle : classes.headerTitle}>
              {this.props.DialogTitle}
            </h6>
            <h5 className={classes.headerID}>{this.props.DialogID === '' ? '' : 'ID:' + this.props.DialogID}</h5>
          </div>
          <div className={classes.dialogBody}>
            <label className={classes.filter}>
              {_.NAME}
              <div>
                <input type='text' className={'pt-input'} ref='serviceName' defaultValue={this.props.DialogName}
                  onChange={this.handleChange('name')} />
              </div>
            </label>
            <label className={classes.filter}>
              {_.KERNELID}
              <div>
                {(() => {
                  if (!this.props.loadingKernelIds) {
                    if (this.props.kernelIds.length) {
                      return (<select
                        className={classes.select}
                        disabled={this.props.DialogBtnContent === 'Edit' && !!this.props.kernelId}
                        onChange={this.handleChange('kernelId')}
                      >
                        {this.props.DialogBtnContent !== 'Edit' ? <option>请选择</option> : ''}
                        {this.props.kernelIds.map((kernelId, i) => {
                          return <option
                            key={i}
                            value={kernelId}
                            selected={this.props.kernelId === kernelId}
                          >{kernelId}</option>
                        })}
                      </select>)
                    } else {
                      return <input type='text' className={'pt-input'} value='没有可用的kernelId' disabled />
                    }
                  } else {
                    return <input type='text' className={'pt-input'} value='loading...' disabled />
                  }
                })()}
              </div>
            </label>
            <label className={classes.filter}>
              {_.MODEL}
              <div>
                {(() => {
                  if (this.props.data && this.props.data.model) {
                    return (
                      <select className={classes.select} onChange={this.handleChange('modelId')}>
                        <option>请选择</option>
                        {this.props.data.model.result.Models.map((item, i) => {
                          return <option key={i} value={item.id}
                            selected={+this.props.modelId === +item.id}>{item.path}</option>
                        })}
                      </select>
                    )
                  }
                })()}
              </div>
            </label>
            <div className={classes.btnGroup}>
              <Button className={classes.btnCancel} onClick={this.props.toggleDialog}>
               Cancel
              </Button>
              <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.props.handleSubmit}
                disabled={!this.isChanged()} >
                {this.props.DialogBtnContent}
              </Button>
            </div>
          </div>
        </Dialog>
      </div>
    )
  }
}

export default DialogFilter
