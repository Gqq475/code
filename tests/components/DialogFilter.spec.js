import React from 'react'
import { DialogFilter } from 'components/DialogFilter/DialogFilter'

// TypeError: Cannot read property 'length' of undefined
describe('(Component) DialogFilter', () => {
  it('render normally', () => {
    const data = {
      kernelIds: [1, 98],
      kernelId: 1
    }
    const wrapper = shallow(
      <DialogFilter {...data} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
