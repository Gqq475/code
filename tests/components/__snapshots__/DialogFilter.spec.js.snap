// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) DialogFilter render normally 1`] = `
<div
  className="DialogFilter-container"
>
  <Dialog
    className="dialog"
    isOpen={false}
  >
    <div
      className="header"
    >
      <h6
        className="headerTitle"
      />
      <h5
        className="headerID"
      >
        ID:undefined
      </h5>
    </div>
    <div
      className="dialogBody"
    >
      <label
        className="filter"
      >
        <div>
          <input
            className="pt-input"
            onChange={[Function]}
            type="text"
          />
        </div>
      </label>
      <label
        className="filter"
      >
        <div>
          <select
            className="select"
            disabled={false}
            onChange={[Function]}
          >
            <option>
              请选择
            </option>
            <option
              selected={true}
              value={1}
            >
              1
            </option>
            <option
              selected={false}
              value={98}
            >
              98
            </option>
          </select>
        </div>
      </label>
      <label
        className="filter"
      >
        <div />
      </label>
      <div
        className="btnGroup"
      >
        <Blueprint.Button
          className="btnCancel"
        >
          Cancel
        </Blueprint.Button>
        <Blueprint.Button
          className="pt-button pt-intent-primary btnEdit"
          disabled={true}
        />
      </div>
    </div>
  </Dialog>
</div>
`;
